#!/usr/bin/env python2

PACKAGE = "zeabus_control"
NODE_NAME = "zeabus_control"
INCLUDE_NAME = "pid_"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

type	= [ "velocity"]
subtype = [ "x" , "y" , "z" , "roll" , "pitch" , "yaw"]
pid		= [ "p" , "i" , "d" ]

for run_type in range( 0 , 1):
	for run_subtype in range( 0 , 6):
		for run_pid in range( 0 , 3):
			gen.add( pid[run_pid]+"_"+subtype[run_subtype]+"_"+type[run_type] , double_t , 0 ,
					 pid[run_pid]+"_"+subtype[run_subtype]+"_"+type[run_type] , 0 , 0 , 5)

gen.add( "x_offset" , double_t , 0 , "x_offset" , 0 , -5 , 5 )
gen.add( "y_offset" , double_t , 0 , "y_offset" , 0 , -5 , 5 )
gen.add( "z_offset" , double_t , 0 , "z_offset" , 0 , -5 , 5 )
gen.add( "roll_offset" , double_t , 0 , "roll_offset" , 0 , -5 , 5 )
gen.add( "pitch_offset" , double_t , 0 , "pitch_offset" , 0 , -5 , 5 )
gen.add( "yaw_offset" , double_t , 0 , "yaw_offset" , 0 , -5 , 5 )

exit( gen.generate( PACKAGE , NODE_NAME , INCLUDE_NAME))
